<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>BioXpress Downloader Step &mdash; HIVE BioXpress v-5.0 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="BioXpress Annotation Step" href="annotation.html" />
    <link rel="prev" title="BioXpress Pipeline v-5.0 README" href="index.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> HIVE BioXpress
          </a>
              <div class="version">
                v-5.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">BioXpress Downloader Step</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#general-flow-of-scripts"><strong>General Flow of Scripts</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#procedure"><strong>Procedure</strong></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#downloader-step-1-get-sample-list-files-from-the-gdc-data-portal"><em>Downloader Step 1</em> : Get sample list files from the GDC Data Portal</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#summary"><em>Summary</em></a></li>
<li class="toctree-l4"><a class="reference internal" href="#method"><em>Method</em></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#downloader-step-2-run-the-script-get-data-all-samples-sh"><em>Downloader Step 2</em>: Run the script get_data_all_samples.sh</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#id1"><em>Summary</em></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id2"><em>Method</em></a></li>
<li class="toctree-l4"><a class="reference internal" href="#output"><em>Output</em></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#downloader-step-3-get-hits-into-dir-py"><em>Downloader Step 3</em>: get_hits_into_dir.py</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#id3"><em>Summary</em></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id4"><em>Method</em></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id5"><em>Output</em></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#downloader-step-4-run-the-script-merge-files-tumor-and-normal-sh"><em>Downloader Step 4</em>: Run the script merge_files_tumor_and_normal.sh</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#id6"><em>Summary</em></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id7"><em>Method</em></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id8"><em>Output</em></a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="annotation.html">BioXpress Annotation Step</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="deseq.html">BioXpress DESeq step</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="publisher.html">BioXpress Publisher Step</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="updates.html">Major Changes from v-4.0</a></li>
<li class="toctree-l1"><a class="reference internal" href="oncomx_glygen.html">Post-processing for OncoMX and Glygen</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">HIVE BioXpress</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
      <li>BioXpress Downloader Step</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/downloader.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="bioxpress-downloader-step">
<h1>BioXpress Downloader Step<a class="headerlink" href="#bioxpress-downloader-step" title="Permalink to this headline"></a></h1>
<p><em>Step 1 of the BioXpress pipeline</em></p>
<p>The downloader step will use sample sheets obtained from <a class="reference external" href="https://portal.gdc.cancer.gov/repository">GDC Data Portal</a> to download raw counts from RNA-Seq for Primary Tumor and Normal Tissue in all available TCGA Studies.</p>
<section id="general-flow-of-scripts">
<h2><strong>General Flow of Scripts</strong><a class="headerlink" href="#general-flow-of-scripts" title="Permalink to this headline"></a></h2>
<p>get_data_all_samples.sh -&gt; get_hits_into_dir.py -&gt; merge_files_tumor_and_normal.sh</p>
</section>
<section id="procedure">
<h2><strong>Procedure</strong><a class="headerlink" href="#procedure" title="Permalink to this headline"></a></h2>
<section id="downloader-step-1-get-sample-list-files-from-the-gdc-data-portal">
<h3><em>Downloader Step 1</em> : Get sample list files from the GDC Data Portal<a class="headerlink" href="#downloader-step-1-get-sample-list-files-from-the-gdc-data-portal" title="Permalink to this headline"></a></h3>
<section id="summary">
<h4><em>Summary</em><a class="headerlink" href="#summary" title="Permalink to this headline"></a></h4>
<p>Sample sheets are downloaded from the GDC data portal and used for the downstream scripts to obtain read count files.</p>
</section>
<section id="method">
<h4><em>Method</em><a class="headerlink" href="#method" title="Permalink to this headline"></a></h4>
<p>Go to the <a class="reference external" href="https://portal.gdc.cancer.gov/repository">GDC Repository</a></p>
<p>Click on the button labeled <code class="docutils literal notranslate"><span class="pre">Advanced</span> <span class="pre">Search</span></code> on the upper right of the repository home page.</p>
<ul class="simple">
<li><p>All of the filters can also be selected manually using the search tree on the left side of the page at the link above. To select a files filter or a cases filter, that tab must be selected on the search bar</p></li>
</ul>
<p>To get the Primary Tumor samples, enter the following into the query box</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">files</span><span class="o">.</span><span class="n">analysis</span><span class="o">.</span><span class="n">workflow_type</span> <span class="ow">in</span> <span class="p">[</span><span class="s2">&quot;HTSeq - Counts&quot;</span><span class="p">]</span>  <span class="ow">and</span> <span class="n">files</span><span class="o">.</span><span class="n">data_type</span> <span class="ow">in</span> <span class="p">[</span><span class="s2">&quot;Gene Expression Quantification&quot;</span><span class="p">]</span>
<span class="ow">and</span> <span class="n">cases</span><span class="o">.</span><span class="n">samples</span><span class="o">.</span><span class="n">sample_type</span> <span class="ow">in</span> <span class="p">[</span><span class="s2">&quot;Primary Tumor&quot;</span><span class="p">]</span> <span class="ow">and</span> <span class="n">cases</span><span class="o">.</span><span class="n">project</span><span class="o">.</span><span class="n">program</span><span class="o">.</span><span class="n">name</span> <span class="ow">in</span> <span class="p">[</span><span class="s2">&quot;TCGA&quot;</span><span class="p">]</span>
</pre></div>
</div>
<p>Click <code class="docutils literal notranslate"><span class="pre">Submit</span> <span class="pre">Query</span></code></p>
<p>The search results screen will now appear. On this screen, click <code class="docutils literal notranslate"><span class="pre">Add</span> <span class="pre">All</span> <span class="pre">Files</span> <span class="pre">To</span> <span class="pre">Cart</span></code>
Then select the <code class="docutils literal notranslate"><span class="pre">Cart</span></code> on the upper right of the page.</p>
<p>Click <code class="docutils literal notranslate"><span class="pre">Sample</span> <span class="pre">Sheet</span></code> from the Cart page to download the Sample Sheet for the Primary Tumor samples.</p>
<ul class="simple">
<li><p>You will need to change the name for the sample sheet, otherwise when we download the sample sheet for Normal tissues it will have the same file name and overwrite the previous file. Add <code class="docutils literal notranslate"><span class="pre">tumor</span></code> or <code class="docutils literal notranslate"><span class="pre">normal</span></code> to the file names when downloaded.</p></li>
</ul>
<p>Remove all samples from the cart, then repeat Step 1 for the Normal Tissue samples.</p>
<p>In the Advanced Search query box add enter the following for Normal Tissue samples:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">files</span><span class="o">.</span><span class="n">analysis</span><span class="o">.</span><span class="n">workflow_type</span> <span class="ow">in</span> <span class="p">[</span><span class="s2">&quot;HTSeq - Counts&quot;</span><span class="p">]</span>  <span class="ow">and</span> <span class="n">files</span><span class="o">.</span><span class="n">data_type</span> <span class="ow">in</span> <span class="p">[</span><span class="s2">&quot;Gene Expression Quantification&quot;</span><span class="p">]</span>
<span class="ow">and</span> <span class="n">cases</span><span class="o">.</span><span class="n">samples</span><span class="o">.</span><span class="n">sample_type</span> <span class="ow">in</span> <span class="p">[</span><span class="s2">&quot;Solid Tissue Normal&quot;</span><span class="p">]</span> <span class="ow">and</span> <span class="n">cases</span><span class="o">.</span><span class="n">project</span><span class="o">.</span><span class="n">program</span><span class="o">.</span><span class="n">name</span> <span class="ow">in</span> <span class="p">[</span><span class="s2">&quot;TCGA&quot;</span><span class="p">]</span>
</pre></div>
</div>
<p>Once both Sample Sheets are downloaded, Primary Tumor and Normal Tissue, move both sample sheets to the server on which the pipeline will be ran, to the path <code class="docutils literal notranslate"><span class="pre">/data/projects/bioxpress/$version/downloads/</span></code> where $version one increment higher then the latest version of BioXpress i.e. latest version is <code class="docutils literal notranslate"><span class="pre">v-4.0</span></code> so new run will be <code class="docutils literal notranslate"><span class="pre">v-5.0</span></code>.</p>
</section>
</section>
<section id="downloader-step-2-run-the-script-get-data-all-samples-sh">
<h3><em>Downloader Step 2</em>: Run the script get_data_all_samples.sh<a class="headerlink" href="#downloader-step-2-run-the-script-get-data-all-samples-sh" title="Permalink to this headline"></a></h3>
<section id="id1">
<h4><em>Summary</em><a class="headerlink" href="#id1" title="Permalink to this headline"></a></h4>
<p>The shell script get_data_all_samples.sh provides arguments to the python script get_data_all_samples.py. It generates a log file that is used to create directories and filter out TCGA studies with low sample numbers.</p>
</section>
<section id="id2">
<h4><em>Method</em><a class="headerlink" href="#id2" title="Permalink to this headline"></a></h4>
<p>Edit the hard-coded paths in the script get_data_all_samples.sh</p>
<blockquote>
<div><ul class="simple">
<li><p>The shell script will call the python script once for the tumor samples and once for the normal sample, so for both tumor and normal you will need to specify the path to the appropriate sample sheet and the path to the log file</p></li>
</ul>
</div></blockquote>
<p>Edit a hard-coded path in the script get_data_all_samples.py</p>
<blockquote>
<div><ul class="simple">
<li><p>Edit the line (~line 44) <code class="docutils literal notranslate"><span class="pre">path0</span> <span class="pre">=</span> <span class="pre">&quot;/data/projects/bioxpress/$version/downloads&quot;</span></code> with the version for your current run of bioxpress.</p></li>
</ul>
</div></blockquote>
<p>Run the shell script <code class="docutils literal notranslate"><span class="pre">sh</span> <span class="pre">get_data_all_samples.sh</span></code></p>
</section>
<section id="output">
<h4><em>Output</em><a class="headerlink" href="#output" title="Permalink to this headline"></a></h4>
<p>After the script has completed, you will have a folder for each TCGA study with all read count files compressed into a file <code class="docutils literal notranslate"><span class="pre">results.tar.gz</span></code>. You will also have three log files, one each for Tumor and Normal as well as a third log file that is the two combined <code class="docutils literal notranslate"><span class="pre">get_data_all_samples.log</span></code></p>
</section>
</section>
<section id="downloader-step-3-get-hits-into-dir-py">
<h3><em>Downloader Step 3</em>: <a class="reference external" href="https://github.com/GW-HIVE/bioxpress/blob/main/downloader/get_hits_into_dir.py">get_hits_into_dir.py</a><a class="headerlink" href="#downloader-step-3-get-hits-into-dir-py" title="Permalink to this headline"></a></h3>
<section id="id3">
<h4><em>Summary</em><a class="headerlink" href="#id3" title="Permalink to this headline"></a></h4>
<p>The python script get_hits_into_dir.py decompresses all read count files and uses the log file generated in the previous script to filter out all TCGA studies that have less than 10 Normal Tissue samples. Count files are generated and labeled as <cite>intermediate</cite> because they will be further manipulated in later Steps</p>
</section>
<section id="id4">
<h4><em>Method</em><a class="headerlink" href="#id4" title="Permalink to this headline"></a></h4>
<p>Edit the hard-coded paths in get_hits_into_dir.py</p>
<blockquote>
<div><ul class="simple">
<li><p>Edit the line (line ~12) <code class="docutils literal notranslate"><span class="pre">with</span> <span class="pre">open(&quot;/data/projects/bioxpress/$version/downloads/get_data_all_samples.log&quot;,</span> <span class="pre">'r')</span> <span class="pre">as</span> <span class="pre">fil:</span></code> with the version for your current run of BioXpress
- Ensure that the log file is the joined log file from the previous script, it should contain information for both Primary Tumor and Solid Tissue normal</p></li>
<li><p>Edit the line (line ~44) <code class="docutils literal notranslate"><span class="pre">topDir</span> <span class="pre">=</span> <span class="pre">&quot;/data/projects/bioxpress/$version/downloads/&quot;</span></code> with the version for your current run of BioXpress</p></li>
</ul>
</div></blockquote>
<p>Run the python script <code class="docutils literal notranslate"><span class="pre">python</span> <span class="pre">get_hits_into_dir.py</span></code></p>
</section>
<section id="id5">
<h4><em>Output</em><a class="headerlink" href="#id5" title="Permalink to this headline"></a></h4>
<p>For each TCGA study there will be a folder named <code class="docutils literal notranslate"><span class="pre">$study_$sampletype_intermediate</span></code> that contains a read count file for each sample within that study.</p>
</section>
</section>
<section id="downloader-step-4-run-the-script-merge-files-tumor-and-normal-sh">
<h3><em>Downloader Step 4</em>: Run the script merge_files_tumor_and_normal.sh<a class="headerlink" href="#downloader-step-4-run-the-script-merge-files-tumor-and-normal-sh" title="Permalink to this headline"></a></h3>
<section id="id6">
<h4><em>Summary</em><a class="headerlink" href="#id6" title="Permalink to this headline"></a></h4>
<p>The shell script merge_files_tumor_and_normal.sh provides arguments to the python script merge_files_tumor_and_normal.py. All read count files for Tumor and Normal from the intermediate folders are merged so that there is one read count file per study (All samples as fields and one row per gene) and one category file per study (defines whether a sample ID corresponds to Primary-Tumor or Solid Tissue Normal).</p>
</section>
<section id="id7">
<h4><em>Method</em><a class="headerlink" href="#id7" title="Permalink to this headline"></a></h4>
<p>Edit the hard-coded paths in merge_files_tumor_and_normal.sh</p>
<ul class="simple">
<li><p>Specify the paths for the variables <code class="docutils literal notranslate"><span class="pre">in_dir</span></code> and <code class="docutils literal notranslate"><span class="pre">out_dir</span></code></p></li>
</ul>
<p>Run the shell script <code class="docutils literal notranslate"><span class="pre">sh</span> <span class="pre">merge_files_tumor_and_normal.sh</span></code></p>
</section>
<section id="id8">
<h4><em>Output</em><a class="headerlink" href="#id8" title="Permalink to this headline"></a></h4>
<p>The <code class="docutils literal notranslate"><span class="pre">out_dir</span></code> specified in merge_files_tumor_and_normal.sh contains two files per study, one for counts and one for categories. The counts files contains all read counts for that study for each gene and provide sample IDs as the fields. The categories file contains information on each sample ID as either Primary Tumor or Solid Tissue Normal.</p>
<p>For checking sample names and numbers lists from v-5.0, all lists and the sample log have been moved to the folder <code class="docutils literal notranslate"><span class="pre">downloads/v-5.0/sample_lists</span></code>.</p>
</section>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="index.html" class="btn btn-neutral float-left" title="BioXpress Pipeline v-5.0 README" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="annotation.html" class="btn btn-neutral float-right" title="BioXpress Annotation Step" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2021, GW HIVE Lab.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>